CRM/Core/Smarty/plugins/modifier.crmMoney.php

```diff
@@ -29,5 +29,8 @@
  * @throws \CRM_Core_Exception
  */
 function smarty_modifier_crmMoney($amount, ?string $currency = NULL): string {
+  if (is_null($amount) || $amount === '') {
```

Comment:
Would it make sense to put this in Civi/Core/Format::money() itself? Because you'll get a similar error thrown from Brick if some php code calls it directly with null or ''.

Reply:
I would :+1: that because it would make Civi less crashy if `Civi/Core/Format::money()` is ever called with empty values, but I'll defer to @eileenmcnaughton if she prefers it to be stricter and is confident that the calling functions would never do that.

Reply:
So I'm pretty confident that functions that already call it would never pass in an empty - I'm on the fence about whether we should permit empties or not

Reply:
But I guess we should accept empty strings - since the typing permits strings - the bigger questions are FALSE or NULL

Reply:
I've created a more forgiving version of this PR at #22429 and I'll let @eileenmcnaughton decide which to merge. My vote would be to merge the more forgiving one because this is a user-facing output function so strictness to the point of a hard-crash is overkill IMO.